# HG changeset patch
# User BelovaMM
# Parent  0d690ab984cb08c5ccc769631210c5fd6e0b7fde
refs #DEVBO-674 ТС ПИоТ ГИС МТ. Отладка. #BTS-2407 #BTS-2408
BTS-2407 При выключенной кассе в ТН не происходит отправка настроек для проверки КМ
BTS-2408 После передачи настроек для проверки КМ из ТН, на текстовой кассе изменяется параметр MACC_IP
https://msk-exp-jira.expertek.local/browse/DEVBO-674
https://msk-exp-jira.expertek.local/browse/BTS-2407
https://msk-exp-jira.expertek.local/browse/BTS-2408

diff --git a/ibs/th/skt/ControlledClients/gismtprop.i b/ibs/th/skt/ControlledClients/gismtprop.i
--- a/ibs/th/skt/ControlledClients/gismtprop.i
+++ b/ibs/th/skt/ControlledClients/gismtprop.i
@@ -103,7 +103,8 @@ end.
             when "logical"
                then return string(buf_thbj-attr.property-value-logical).         
         end case.     
-    end.       
+    end. 
+    return "".      
 end.
 
 { def/funcmet.i get-gismt-prop character }
diff --git a/str/send-all-work-gismt.p b/str/send-all-work-gismt.p
--- a/str/send-all-work-gismt.p
+++ b/str/send-all-work-gismt.p
@@ -188,6 +188,8 @@ procedure putc :
          
    end.
    if not vAll then do:
+       if vTH_IP = ? then vTH_IP = "".
+       if vTH_Port = ? then vTH_Port = "".
        /* возможно изменили только одно из двух полей, надо тогда вычислить MACC_IP */
        if vTH_IP = "" and vTH_Port <> "" then vTH_IP = get-thbj-attr-prop({&db},mdb-num,{&attr-gisMT},{&attr-gisMT_TH_IP}).
        if vTH_IP <> "" and vTH_Port = "" then vTH_Port = get-thbj-attr-prop({&db},mdb-num,{&attr-gisMT},{&attr-gisMT_TH_Port}).
@@ -204,7 +206,7 @@ procedure putc :
        if vMACC_IP <> "" and vLmCHzPort <> "" then
          run put-xml-data(iSAXWriter,"LmCHzPort",vLmCHzPort,"Порт для отправки запроса проверки марки в ЛМ ЧЗ ").  
    end.    
-   else do: 
+   else do:           
       assign
         vCheckBlock     = get-list-code-typemark(get-thbj-attr-prop(mObjType,mObjCode,{&attr-marking},{&attr-marking_checkBlock}))        
         vCheckDate      = get-list-code-typemark(get-thbj-attr-prop(mObjType,mObjCode,{&attr-marking},{&attr-marking_checkDate}))                                            
@@ -221,7 +223,10 @@ procedure putc :
         vTH_Port = get-thbj-attr-prop({&db},mdb-num,{&attr-gisMT},{&attr-gisMT_TH_Port})
         vLmCHzPort = get-thbj-attr-prop({&db},mdb-num,{&attr-gisMT},{&attr-gisMT_LmCHzPort})        
         vAddTimeoutPIoT = DEC(get-thbj-attr-prop({&db},mdb-num,{&attr-gisMT},{&attr-gisMT_AddTimeoutPIoT}))    
-        .  
+        no-error.  
+      if vTH_IP = ? then vTH_IP = "".
+      if vTH_Port = ? then vTH_Port = "".
+      if vLmCHzPort = ? then vLmCHzPort = "".
             
       if vTH_IP <> "" and vTH_Port <> "" then vMACC_IP = substitute("&1:&2",vTH_IP,vTH_Port).
       if vCheckBlock <> ? then
diff --git a/str/send-all.p b/str/send-all.p
--- a/str/send-all.p
+++ b/str/send-all.p
@@ -128,17 +128,31 @@ no-error
 .
 if i-Type eq "?"
 then
-   return error "Незадан тип для send-all.p".
+   return error "Не задан тип для send-all.p".
 define variable mValue   as character no-undo.
 define variable mNumPar  as integer no-undo.
 define variable mSendAll as logical no-undo.
 define variable mCashNum as integer no-undo init ?.
+define variable mCashAll as logical no-undo.
+define variable mNumCashAll as integer no-undo.
 mValue = replace(i-value,",","=").
 mNumPar = lookup("cash-send",mValue,"=").
+
+
 if mNumPar > 0
 then do:
    mSendAll = entry(mNumPar + 1,mValue,"=") eq "all" no-error.
    mCashNum = int(entry(mNumPar + 1,mValue,"=")) no-error.
+   if mCashNum <> ? and not mSendAll then do:
+      /* проверяем параметр, надо ли посылать на указанный номер кассы, даже если она 
+      ** выключена в справочнике касс */
+      mNumCashAll = lookup("cash-all",mValue,"="). 
+      if mNumCashAll > 0 then do: 
+          mCashAll = entry(mNumCashAll + 1,mValue,"=") eq "all" no-error.
+          if mCashNum <> ? and mCashNum <> 0 and mCashAll 
+          then mSendAll = yes.
+      end.
+   end.   
 end.
 mNumPar = lookup("SocetLog",mValue,"=").
 if mNumPar > 0
diff --git a/utl/proc-cashsend.p b/utl/proc-cashsend.p
--- a/utl/proc-cashsend.p
+++ b/utl/proc-cashsend.p
@@ -90,7 +90,7 @@ else if mobjsend eq "gismt" or mobjsend 
         &endif
       , input ?
       , input "str/sendgismt.p":U
-      , input substitute ("&2&1&3&1&4&1&5&1&6&1cash-send=&7,&8", 
+      , input substitute ("&2&1&3&1&4&1&5&1&6&1cash-send=&7,&8,&9", 
                                    {&delim-par}, 
                                    {&shop}, 
                                    mshop,
@@ -98,7 +98,8 @@ else if mobjsend eq "gismt" or mobjsend 
                                    "gismt",
                                    'Передача настроек для проверки КМ':U,
                                    mCashNum,
-                                   "SocetLog=" + mSocetLog
+                                   "SocetLog=" + mSocetLog,
+                                   "cash-all=all"
                                    )
       , input ? /*p-auto-go*/
       , input "":U
